src/goban.rs

```diff
@@ -49,26 +49,50 @@ impl Goban
 		((self.enemy | self.player) + Direction::All) & self.list_moves()
 	}
 
+	fn check_surround(&self, lines: BitBoard, dir: Direction) -> u8
+	{
+		let mut ret: u8 = 0;
+		for dirs in [dir, dir.to_invert()].iter()
+		{
+			if ((lines >> *dirs) & self.enemy).is_empty() {
+				ret += 1;
+			}
+		}
+		ret
+	}
+
+	// TODO Add way to isolate different lines, currently cannot differentiate between lines that are on the same axes
 	fn line_detection(&self) -> u16
 	{
 		let mut bits: BitBoard;
+		let mut final_line: BitBoard;
 		let mut total: u16 = 0;
 		let mut len: u16;
 
 		for dir in AxisIterator::new()
 		{
 			bits = self.player;
+			final_line = BitBoard::empty();
 			len = 0;
 			while !bits.is_empty()
```

Comment:
There is now a method called `is_any()` which does exactly the same as `!bits.is_empty()` but without the leading `!` if you want to use it (not necessary here and now, this is mostly advice for the future 🙂 )